### PR TITLE
correctly remove scalar bar actor

### LINKED ISSUE #1368: remove_scalar_bar doesn't release the slot in _scalar_bar_slots and the slot limit can be reached
When using the method `remove_scalar_bar`, the code goes there:
https://github.com/pyvista/pyvista/blob/a3711401a7bd37f655fc346a558f7c67cb522eab/pyvista/plotting/scalar_bars.py#L37-L57

but `self._scalar_bar_mappers` is already empty so it never goes into the for loop and cannot reach the line `self._plotter._scalar_bar_slots.add(slot)` where it should release the slot.

the set of available slots follows this pattern: `{4, 5, 6, 7, 8, 9}` with always only one scalar_bar available and at some point it reaches the limit.

What is the expected behavior? What if there is one or many meshes using that scalar_bar when using that method?

P.S. I just tried overwriting multiple scale_bars with multiple meshes, trying to break it and it works as intented as long as I don't use `remove_scalar_bar` explicitely.

### PR DESCRIPTION
**Pull request by akaszynski**
Resolve #1368 by removing the scalar bar slot when removing the scalar bar.

### CODE REVIEW

**Review by tkoyama010 — APPROVED** [review]
LGTM

File: pyvista/plotting/scalar_bars.py [hunk 1]
```diff
@@ -35,16 +35,23 @@ def __repr__(self):
...
+
+        """
         try:
             mapper = actor.GetMapper()
         except AttributeError:
             return
-        for name in list(self._scalar_bar_mappers):
+
+        for name in self._scalar_bar_mappers:
             try:
```
Comment (adeak) [inline pyvista/plotting/scalar_bars.py:50]: Not sure if you remember, but we talked about this in https://github.com/pyvista/pyvista/pull/1347#pullrequestreview-674843704 and surrounding discussion. Just making sure that this change is intentional in light of new information.

**Comment by adeak** [comment]
OK then 😆

**Comment by akaszynski** [comment]
Sorry, on mobile and didn't see your comment.  I'll open another PR @adeak

**Comment by adeak** [comment]
No worries @akaszynski :) And only open the PR if you think it was better off originally.

**Comment by akaszynski** [comment]
> And only open the PR if you think it was better off originally.

You're correct that the number of keys will change during the loop.  While going through the fix I made the change without thinking about the consequences (figured that someone just made a mistake turning the keys into a list).  I'll add a note to prevent this.

File: pyvista/plotting/scalar_bars.py (hunk 1 above)
Comment (akaszynski) [reply pyvista/plotting/scalar_bars.py:50]: See #1554

**Pull request merged**